twitter/views.py

```diff
@@ -78,62 +78,62 @@ def signup(request):
-#             result = Connection.objects.filter(follower__username=context['user'].username).filter(following__username=username)
-#             context['connected'] = True if result else False
-
-#         return context
-
-# @login_required
-# def follow_view(request, *args, **kwargs):
-#     try:
-#         follower = User.objects.get(username=request.user)
-#         following = User.objects.get(username=kwargs['username'])
-#     except User.DoesNotExist:
-#         messages.warning(request, '{}は存在しません'.format(kwargs['username']))
-#         return HttpResponseRedirect(reverse_lazy('twitter:index'))
-
-#     if follower == following:
-#         messages.warning(request, '自分自身はフォローできませんよ')
-#     else:
-#         _, created = Connection.objects.get_or_create(follower=follower, following=following)
-
-#         if (created):
-#             messages.success(request, '{}をフォローしました'.format(following.username))
-#         else:
-#             messages.warning(request, 'あなたはすでに{}をフォローしています'.format(following.username))
-
-#     return HttpResponseRedirect(reverse_lazy('twitter:profile', kwargs={'username': following.username}))
-
-# @login_required
-# def unfollow_view(request, *args, **kwargs):
-#     try:
-#         follower = User.objects.get(username=request.user)
-#         following = User.objects.get(username=kwargs['username'])
-#         if follower == following:
-#             messages.warning(request, '自分自身のフォローを外せません')
-#         else:
-#             unfollow = Connection.objects.get(follower=follower, following=following)
-#             unfollow.delete()
-#             messages.success(request, 'あなたは{}のフォローを外しました'.format(following.username))
-#     except User.DoesNotExist:
-#         messages.warning(request, '{}は存在しません'.format(kwargs['username']))
-#         return HttpResponseRedirect(reverse_lazy('twitter:index'))
-#     except Connection.DoesNotExist:
-#         messages.warning(request, 'あなたは{0}をフォローしませんでした'.format(following.username))
-
-#     return HttpResponseRedirect(reverse_lazy('twitter:profile', kwargs={'username': following.username}))
\ No newline at end of file
+class ProfileDetailView(LoginRequiredMixin, DetailView):
+     model = User
+     template_name = "twitter/profile.html"
+     slug_field = 'username'
+     slug_url_kwarg = 'username'
+
+     def get_context_data(self, **kwargs):
+         context = super(ProfileDetailView, self).get_context_data(**kwargs)
+         username = self.kwargs['username']
+         context['username'] = username
+         context['user'] = get_current_user(self.request)
+         context['following'] = Connection.objects.filter(follower__username=username).count()
+         context['followers'] = Connection.objects.filter(following__username=username).count()
+
+         if username is not context['user'].username:
+             result = Connection.objects.filter(follower__username=context['user'].username).filter(following__username=username)
```

Comment:
以下のようにまとめて書けないか検証してもらえますか？
```suggestion
             result = Connection.objects.filter(follower__username=context['user'].username, following__username=username)
```